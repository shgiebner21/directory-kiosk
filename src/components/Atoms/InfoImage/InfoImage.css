.image {
  position: relative;
  display: inline-block;
  /* background-color: var(--brandcolor); */
  color: white;
}

.image:after {
  box-shadow: 0px 0px 4px 0px rgba(0, 0, 0, 0.55) inset;
  content: '';
  display: block;
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
}

.image-img {
  background-size: cover;
  z-index: -1;
}
