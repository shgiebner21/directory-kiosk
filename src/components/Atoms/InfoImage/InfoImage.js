import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './InfoImage.css'
// import { getInitials } from '../../../Helpers/Helpers'
import profile_placeholder from '../../../Images/profile_placeholder.png'

/**
 * An avatar buggle that can show a users profile picture
 * or placeholder image.
 */
export default class InfoImage extends Component {
  static propTypes = {
    /** Size for the Image */
    size: PropTypes.oneOf([
      'small',
      'medium',
      'large',
      'big',
      'huge',
      'massive',
    ]),
    /** url of image to be displayed */
    src: PropTypes.string,
    /** Can be dimmed */
    dimmed: PropTypes.bool,
  }
  static defaultProps = {
    size: 'medium',
    src: '',
    dimmed: false,
  }

  constructor(props) {
    super(props)
    this.state = {
      someState: true,
    }
  }

  render() {
    const { size, src, dimmed } = this.props

    const pixelSizes = {
      small: 24,
      medium: 32,
      large: 64,
      big: 96,
      huge: 128,
      massive: 240,
    }

    let sizePx = pixelSizes[size]

    const style = {
      lineHeight: sizePx + 'px',
      width: sizePx + 'px',
      height: sizePx + 'px',
    }

    if (dimmed) style.opacity = 0.5

    const srcOverride = src === '' ? profile_placeholder : src

    const wrapperStyle = Object.assign({}, style, this.props.style)

    return (
      <div className="image" style={wrapperStyle}>
        <img className="image-img" src={srcOverride} alt="" style={style} />
      </div>
    )
  }
}
