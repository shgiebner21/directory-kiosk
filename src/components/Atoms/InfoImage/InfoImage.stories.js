import React from 'react'

import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import centered from '@storybook/addon-centered'
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs'

import InfoImage from './InfoImage'

const stories = storiesOf('Atoms', module)
stories.addDecorator(withKnobs)

stories.addDecorator(centered).add(
  'InfoImage',
  withInfo()(() => {
    const sizeLabel = 'Size'
    const options = {
      small: 'small',
      medium: 'medium',
      large: 'large',
      big: 'big',
      huge: 'huge',
      massive: 'massive',
    }
    const defaultValue = 'medium'
    const value = select(sizeLabel, options, defaultValue)

    const textLabel = 'Name'
    const textDefaultValue = 'Korben Dallas'
    const textValue = text(textLabel, textDefaultValue)

    const dimmedLabel = 'dimmed'
    const dimmedDefaultValue = false
    const dimmedValue = boolean(dimmedLabel, dimmedDefaultValue)

    const picLabel = 'Show picture'
    const picDefaultValue = true
    const picValue = boolean(picLabel, picDefaultValue)

    const picSource = picValue
      ? 'https://tinyfac.es/data/avatars/852EC6E1-347C-4187-9D42-DF264CCF17BF-200w.jpeg'
      : ''

    return (
      <InfoImage
        name={textValue}
        size={value}
        src={picSource}
        dimmed={dimmedValue}
      />
    )
  })
)
