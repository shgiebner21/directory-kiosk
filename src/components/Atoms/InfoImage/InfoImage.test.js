import React from 'react'
import InfoImage from './InfoImage'

describe('InfoImage', () => {
  const userName = 'Korben Dallas'
  const size = 'small'
  const src = 'https://www.fillmurray.com/60/60'
  const sizes = {
    small: '24px',
    medium: '32px',
    large: '64px',
    big: '96px',
    huge: '128px',
    massive: '160px',
  }

  it('matches the snapshot', () => {
    const wrapper = window.shallow(<InfoImage size={size} src={src} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot when dimmed', () => {
    const wrapper = window.shallow(
      <InfoImage size={size} src={src} dimmed={true} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with an image', () => {
    const wrapper = window.shallow(<InfoImage size={size} src={src} />)
    expect(wrapper.find('img')).toBeTruthy()
  })

  it('uses the small size prop', () => {
    const wrapper = window.shallow(<InfoImage size={'small'} src={src} />)
    expect(wrapper.find('div.image').prop('style')).toHaveProperty(
      'width',
      sizes.small
    )
    expect(wrapper.find('div.image').prop('style')).toHaveProperty(
      'height',
      sizes.small
    )
  })

  it('uses the medium size prop', () => {
    const wrapper = window.shallow(<InfoImage size={'medium'} src={src} />)
    expect(wrapper.find('div.image').prop('style')).toHaveProperty(
      'width',
      sizes.medium
    )
    expect(wrapper.find('div.image').prop('style')).toHaveProperty(
      'height',
      sizes.medium
    )
  })

  it('uses the large size prop', () => {
    const wrapper = window.shallow(<InfoImage size={'large'} src={src} />)
    expect(wrapper.find('div.image').prop('style')).toHaveProperty(
      'width',
      sizes.large
    )
    expect(wrapper.find('div.image').prop('style')).toHaveProperty(
      'height',
      sizes.large
    )
  })

  it('uses the big size prop', () => {
    const wrapper = window.shallow(<InfoImage size={'big'} src={src} />)
    expect(wrapper.find('div.image').prop('style')).toHaveProperty(
      'width',
      sizes.big
    )
    expect(wrapper.find('div.image').prop('style')).toHaveProperty(
      'height',
      sizes.big
    )
  })

  it('uses the huge size prop', () => {
    const wrapper = window.shallow(<InfoImage size={'huge'} src={src} />)
    expect(wrapper.find('div.image').prop('style')).toHaveProperty(
      'width',
      sizes.huge
    )
    expect(wrapper.find('div.image').prop('style')).toHaveProperty(
      'height',
      sizes.huge
    )
  })

  it('uses the massive size prop', () => {
    const wrapper = window.shallow(<InfoImage size={'massive'} src={src} />)
    expect(wrapper.find('div.image').prop('style')).toHaveProperty(
      'width',
      sizes.massive
    )
    expect(wrapper.find('div.image').prop('style')).toHaveProperty(
      'height',
      sizes.massive
    )
  })
})
