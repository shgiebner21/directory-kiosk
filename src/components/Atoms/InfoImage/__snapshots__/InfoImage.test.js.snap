// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoImage matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`InfoImage matches the snapshot when dimmed 1`] = `ShallowWrapper {}`;
